lib/tasks/analytics.rake

```diff
@@ -0,0 +1,102 @@
+# New taks should be added here when we add a new table which needs to have data analysed.
+# Upload csv files to Google storage from DB
+namespace :db do
+  namespace :analytics do
+    desc 'Users table'
+    task users: :environment do
+      # due to the nature of the dynamic json objects stored in the column  we need to get the largest column data and run through
+      # AnalyticsBuild::build_json_sql function to create a sql which can be used to populate csv files.
+      # we use COALESCE function to add null value as currently we find on occassions google data studio struggles with empty values.
+
+      # REMOVE THESE: once we fix the data dashboard to match dynamic keys we can remove these from sql
+      # COALESCE(module_time_to_completion->>'child-development-and-the-eyfs', 'null') AS module_1_time,
+      # COALESCE(module_time_to_completion->>'brain-development-and-how-children-learn', 'null') AS module_2_time,
+      # COALESCE(module_time_to_completion->>'personal-social-and-emotional-development', 'null') AS module_3_time,
+      # COALESCE(module_time_to_completion->>'module-4', 'null') AS module_4_time,
+
+      sql = 'SELECT id, module_time_to_completion as json_column, (SELECT COUNT(*) FROM jsonb_object_keys(module_time_to_completion)) nbr_keys FROM public.users order by nbr_keys desc limit 1'
+      user_json = ActiveRecord::Base.connection.execute(sql)
+      users_all = User.select(" id,
```

Comment:
The transformation of this data is implemented in a rather complex way and we do not test the resulting output.